src/test/transport.rs

```diff
@@ -1,15 +1,102 @@
-//! Implementation of a transport for testing purposes. This is largely based on
-//! the `rust-web3` `TestTransport` type with some modifications.
+//! Implementation of a transport for testing purposes.
 
 use jsonrpc_core::{Call, Value};
 use std::cell::RefCell;
+use std::clone::Clone;
 use std::collections::VecDeque;
+use std::fmt::Debug;
 use std::rc::Rc;
 use web3::error::Error;
-use web3::futures::future::{self, FutureResult};
-use web3::helpers;
+use web3::futures::future::FutureResult;
 use web3::{RequestId, Transport};
 
+// ----------------------------------------------------
+
+/// Request
+#[derive(Debug)]
+pub struct Request {
+    pub method: String,
+    pub params: Vec<Value>,
+}
+
+/// RequestResponse
+#[derive(Debug)]
+pub struct RequestResponse {
+    request: Request,
+    response: Result<Value, Error>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct TestTransport_ {
```

Comment:
Why _ at the end? 

Also I think technically this is a "Fake"Transport according to [this categorization](https://martinfowler.com/bliki/TestDouble.html)

Reply:
This PR is to show how the result looks. To make the code compile I need to keep the old TestTransport because I have only converted 2 tests so far.

Reply:
I would argue that this case it is a `MockTransport` and not a `FakeTransport` as the implementation is purely for encoding expectations and not providing a simplified implementation (like the in memory database in his example).